Dockerfile

```diff
@@ -0,0 +1,41 @@
+# This is used to compile the client-side application.
+FROM node:10 as build
+
+ENV NODE_ENV production
+
+WORKDIR /app
+
+# Install the project's dependencies, including dev dependencies
+COPY package.json yarn.lock ./
+RUN yarn install --pure-lockfile --production=false
+
+# Build the client-side application.
+COPY . .
+RUN yarn build
+
+# Build the final Docker image that runs the server and serves the client-side
+# application.
+FROM node:10 as runtime
+
+ENV NODE_ENV production
+ENV PORT 4000
+
+WORKDIR /app
+
+# Install the project's dependencies
+COPY package.json yarn.lock ./
+RUN yarn install --pure-lockfile --production=true
+
+# Add the compiled client-side application
+COPY --from=build /app/build/ ./build
+# Add the server code
+COPY --from=build /app/src/server/ ./src/server
```

Comment:
I'm slightly concerned that this will be hard to maintain: if we add new files outside of this path for any reason, we'll probably forget to update the Docker image.

An alternative would be to copy everything from `build` and just remove what we don't need, like `node_modules`. Did you consider that?

Reply:
I like to bundle only what's used so I did not consider copying all the files. I don't really buy the maintainability issue because if, for some reason, we need more files, it means we need a build step for the server code as well.

Reply:
Yes, well, if we were to adjust the build step for the server then we'd have to remember to update this Dockerfile, too. That won't be obvious and nothing will fail. I guess we just need automated functional tests that run from the docker container to make it fail. 

I think the fragility of explicitly allowing select files isn't worth the security benefit but I don't want to hold up the patch. Ops may have more change requests when they look into this, too.